Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +352,869 @@ public override bool IsServerConnected()
+                    if (!string.IsNullOrEmpty(activitiesGroup.ExternalID))
+                    {
+                        UpdateTestCase(activitiesGroup, fatherId, testCaseFields, step);
+                    }
+                    else
+                    {
+                        CreateNewTestCase(activitiesGroup, fatherId, testCaseFields, step);
+                    }
+                }
+                return true;
+            }
+            catch (Exception ex)
+            {
+                result = "Unexpected error occurred- " + ex.Message;
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to export the " + GingerDicser.GetTermResValue(eTermResKey.ActivitiesGroup) + " to QC/ALM", ex);
+                return false;
+            }
+        }
+
+        private string CreateNewTestCase(ActivitiesGroup activitiesGroup, string fatherId, ObservableList<ExternalItemFieldBase> testCaseFields, string testScript)
+        {
+            //set item fields
+            TestManual test = new TestManual();
+            test.Name = activitiesGroup.Name;
+            test.SetValue("description", activitiesGroup.Description);
+            test.SetValue("automation_status", new BaseEntity()
+            {
+                TypeName = "list_node",
+                Id = "list_node.automation_status.not_automated"
+            });
+
+            test.SetValue("product_areas", new EntityList<BaseEntity>()
+            {
+                data = new List<BaseEntity>()
+                {
+                    new BaseEntity() {Id = fatherId, TypeName = "product_area"}
+                }
+            });
+            test.SetValue("test_type", new EntityList<BaseEntity>()
+            {
+                data = new List<BaseEntity>()
+                {
+                    new BaseEntity()
+                    {
+                        TypeName = "list_node",
+                        Id = "list_node.test_type.regression"
+                    }
+                }
+            });
+
+            //foreach (ExternalItemFieldBase field in testCaseFields)
+            //{
+            //    if ((field.ToUpdate || field.Mandatory) && !test.Contains(field.ExternalID))
+            //    {
+            //        if (string.IsNullOrEmpty(field.SelectedValue) == false && field.SelectedValue != "NA")
+            //            test.SetValue(field.ExternalID, field.SelectedValue);                    
+            //    }
+            //}
+
+            test = Task.Run(() => { return this.octaneRepository.CreateEntity(GetLoginDTO(), test, null); }).Result;
+
+            activitiesGroup.ExternalID = test.Id.ToString();
+            activitiesGroup.ExternalID2 = test.Id.ToString();
+            CreateTestStep(test.Id, testScript);
+
+            return test.Id.ToString();
+        }
+
+        private string UpdateTestCase( ActivitiesGroup activitiesGroup, string fatherId, ObservableList<ExternalItemFieldBase> testCaseFields, string testScript)
```

Comment:
Codacy found an issue: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)